### DISCUSSION TITLE
Section Icons (md-nav__link)

### DISCUSSION THREAD

**Discussion by polymorcodeus** [squidfunk/mkdocs-material#4668] [Q&A] [answered]
Heya-

I'm trying to figure out what's the best way to add "section" icons, similar to how page icons are rendered via the front matter [here](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-icon)

If you look to the right of the nav item, you can see the expansion, but for a large side menu calling out those section items - particularly with a quick icon would be useful. Random markup below of the Nav.

```
ACME Corp
 Product One
   Page One
   Page Two
   💯 Section One                             >
         Product One Page all about 100
   👍 Section Two                             >
         Product One Page all about thumbs
  Product Two
   Page One
   Page Two
   💯 Section One                             >
         Prodcut Two Page all about 100
   👍 Section Two                             >
         Product Two Page all about thumbs
```
Adding the Section icon then would do a couple things, allow tagging of similar sections in different 1st level sections W/O having to look at tags of the nested pages, highlight the expandable sections when reading the nav panel left to right, and in my case distinguish the sections from page links (provided I don't supply page icons)

Looking through the partials and how the page.meta items are rendered - topically adding something like this:

```
{% if nav_item.is_section and nav_item.meta.section_icon %}
    {% include ".icons/" ~ nav_item.meta.section_icon ~ ".svg" %}
{% endif %}
```
But then you need a way to give the `mkdocs.structure.nav.Section object` a meta attribute - which doesn't exist like it does for pages. 

I'm hoping someone else has blazed this trail before - but if not, looks like it'll be me.

**Comment by squidfunk** [comment, marked as answer]
Thanks for suggesting. You're right, section icons are not supported, because it's not possible to add metadata to sections. You could solve it by enabling [`navigation.indexes`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#section-index-pages), and add a section index page. However, this would mean, that there would be a new page. Unfortunately, defining front matter for sections is not supported by MkDocs. There surely is a way, but it's going to be hacky.

**Reply by polymorcodeus** [reply]
Yeah hacky seemed like the path forward - I was hoping I was missing something.
